datacapturing/src/main/java/de/cyface/datacapturing/DataCapturingService.java

```diff
@@ -1,85 +1,146 @@
  * before transmitting any data. You may however force synchronization if required, using
  * {@link #forceSyncUnsyncedMeasurements()}.
- * </p>
  * <p>
  * An object of this class is not thread safe and should only be used once per application. You may start and stop the
  * service as often as you like and reuse the object.
- * </p>
  *
  * @author Klemens Muthmann
  * @version 1.0.0
  * @since 1.0.0
  */
 public final class DataCapturingService {
 
+    /*
+     * MARK: Properties
+     */
+
     /**
-     * <p>
      * {@code true} if data capturing is running; {@code false} otherwise.
-     * </p>
      */
     private boolean isRunning = false;
     /**
-     * <p>
      * A listener that is notified of important events during data capturing.
-     * </p>
      */
     private DataCapturingListener listener;
     /**
-     * <p>
      * A poor mans data storage. This is only in memory and will be replaced by a database on persistent storage.
-     * </p>
      */
     private final List<Measurement> unsyncedMeasurements;
+    /**
+     * A weak reference to the calling context. This is a weak reference since the calling context (i.e.
+     * <code>Activity</code>) might have been destroyed, in which case there is no context anymore.
+     */
+    private final WeakReference<Context> context;
+    /**
+     * Connection used to communicate with the background service
+     */
+    private final ServiceConnection serviceConnection;
+    /**
+     * Messenger that handles messages arriving from the <code>DataCapturingBackgroundService</code>.
+     */
+    private Messenger fromServiceMessenger;
+    /**
+     * Messenger used to send messages from this class to the <code>DataCapturingBackgroundService</code>.
+     */
+    private Messenger toServiceMessenger;
 
     /**
-     * <p>
      * Creates a new completely initialized {@link DataCapturingService}.
-     * </p>
+     *
+     * @param context The context (i.e. <code>Activity</code>) handling this service.
      */
-    public DataCapturingService() {
-        unsyncedMeasurements = new ArrayList<>();
+    public DataCapturingService(final Context context) {
+        this.unsyncedMeasurements = new ArrayList<>();
+        this.context = new WeakReference<Context>(context);
+
+        this.serviceConnection = new BackgroundServiceConnection();
     }
 
     /**
-     * <p>
      * Starts the capturing process. This operation is idempotent.
-     * </p>
+     * <p>
+     * Since the method is synchronized with the actual service startup it should be considered a long running operation
+     * and not run on the UI thread to avoid ANR errors.
      */
     public void start() {
-        isRunning = true;
-        unsyncedMeasurements.add(new Measurement(unsyncedMeasurements.size()));
+        // TODO this will cause an error in the handler which currently does not accept null.
+        start(null);
```

Comment:
wip?

Reply:
Jep. Already fixed in the most recent version.